avm/res/insights/metric-alert/main.bicep

```
@@ -203,3 +192,20 @@ type roleAssignmentType = {
   @description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+@discriminator('odata.type')
+type alertType = alertWebtestType | alertResourceType | alertMultiResourceType
+type alertResourceType = {
+  'odata.type': 'Microsoft.Azure.Monitor.SingleResourceMultipleMetricCriteria'
+  allof: array
+}
+type alertMultiResourceType = {
+  'odata.type': 'Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria'
+  allof: array
+}
+type alertWebtestType = {
+  'odata.type': 'Microsoft.Azure.Monitor.WebtestLocationAvailabilityCriteria'
+  componentId: string
+  failedLocationCount: int
+  webTestId: string
```

```suggestion
  componentResourceId: string
  failedLocationCount: int
  webTestResourceId: string
```

Should be this as per spec. However, I'm not sure how this would translate to the mapping in line 132. I guess it would require a union() statement like
```bicep
criteria: union({
  'odata.type': criteria.'odata.type'
},(contains(criteria, 'allof') ? { allof: criteria.allof } : {}),
  (contains(criteria, 'componentResourceId') ? { componentId: criteria.componentResourceId} : {}),
  (contains(criteria, 'failedLocationCount') ? { failedLocationCount: criteria.failedLocationCount} : {}),
  (contains(criteria, 'webTestResourceId') ? { webTestId: criteria.webTestResourceId} : {})
}
```

---

If applied, would require an update to the `main.json`, the `main.test.bicep` files & `readme.md`. Also, it should be tested to ensure it works

---

If applied, would require an update to the main.json, the main.test.bicep files & readme.md. Also, it should be tested to ensure it works

---

Unfortunately, bicep sees the 'odata.type' as a duplicate value in the mapping in line 132 so this option isn't going to work. I am not sure it makes sense to make the change to the discriminator to just have to change the values back to what they are currently set to.

---

Probably not. Let me take a closer look tomorrow, and if there is no easy solution (and I guess there rarely is), I'll resolve the comment.

---

Hey @arnoldna, applied a small update to the example above. Just successfully tested it :)